cfpb/consumerfinance.gov
# Provide a suggestion only when no search results are found

Fixed bug in ask-cfpb search with ES7, it is always suggesting a term even when it has results for said term.

Changed the ES7 suggestion logic in `ask_search_es7` code in the following file:
https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/ask_cfpb/views.py#L147

The issues are as follows:

1. we're always providing the suggestion unless its already been clicked
2. we shouldn't provide a suggestion unless no results are found


## Testing

This app is best tested in docker development mode,
so that Elasticsearch 2 and 7 can be run side by side.
- Pull in the branch and either refresh your database or run `migrate`

```
docker-compose exec python bash

# AND

./refresh-data.sh

# OR

./cfgov/manage.py migrate`
```

- Open the Wagtail admin locally and enable the `ELASTICSEARCH_DSL_ASK` flag at http://localhost:8000/admin/flags/ELASTICSEARCH_DSL_ASK/

- Back in the python container, build the ASK Elasticsearch7 index:

```
./cfgov/manage.py search_index --rebuild -f --models ask_cfpb
```

You could also run the Elasticsearch2 index with

```
./cfgov/manage.py update_index ask_cfpb
```

But it's just as easy to skip that and compare the local activities search to the live one at  
<https://www.consumerfinance.gov/ask-cfpb/search/>

## How to test this PR without using Docker

1. Navigate to https://localhost:8000/ask-cfpb/search
1. Search for something that exists and confirm a term is not suggested
1. Search for something that does not exist and confirm a term is suggested
1. tox -e unittest-current ask_cfpb.tests.test_search.AnswerPageSearchTest


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
